docs/deployment/gae.rst

```diff
@@ -1,225 +1,104 @@
-
-      import sys
-      sys.path = [path for path in sys.path if 'site-packages' not in path]
-      import site
-
-   You will also need to comment out the line that starts with
-   ``assert sys.path`` in the file::
-
-      # comment the sys.path assertion out
-      # assert sys.path[:len(cur_sys_path)] == cur_sys_path, (
-      #   "addsitedir() caused entries to be prepended to sys.path")
-
-   For GAE development environment 1.3.0 or better, you will also need
-   the following somewhere near the top of the ``runner.py`` file to
-   fix a compatibility issue with ``appengine-monkey``::
-
-      import os
-      os.mkdir = None
-
-#. Run the application.  ``dev_appserver.py`` is typically installed
-   by the SDK in the global path but you need to be sure to run it
-   with Python 2.5 (or whatever version of Python your GAE SDK
-   expects).
+system. This tutorial also assumes you've already installed and created a :app:`Pyramid` application,
+and that you have an App Engine account to deploy to.
 
-   .. code-block:: text
-      :linenos:
+Setup
+-----
 
-      $ cd ../..
-      $ python2.5 /usr/local/bin/dev_appserver.py pyramidapp/app/
+   First we'll need to create a few files so that App Engine can communicate with our project properly.
 
-   Startup success looks something like this:
+   Create a ``requirements.txt`` file, a ``main.py`` file, an ``app.yaml`` file, and an ``appengine_config.py`` file.
 
-   .. code-block:: text
+#. Edit ``requirements.txt``
 
-      [chrism@vitaminf pyramid_gae]$ python2.5 \
-                    /usr/local/bin/dev_appserver.py \
-                    pyramidapp/app/
-      INFO     2009-05-03 22:23:13,887 appengine_rpc.py:157] # ... more... 
-      Running application pyramidapp on port 8080: http://localhost:8080
+    Add the following lines:
 
-   You may need to run "Make Symlinks" from the Google App Engine
-   Launcher GUI application if your system doesn't already have the
-   ``dev_appserver.py`` script sitting around somewhere.
+    .. code-block:: text
+        :linenos:
 
-#. Hack on your pyramid application, using a normal run, debug, restart
-   process.  For tips on how to use the ``pdb`` module within Google
-   App Engine, `see this blog post
-   <http://jjinux.blogspot.com/2008/05/python-debugging-google-app-engine-apps.html>`_.
-   In particular, you can create a function like so and call it to
-   drop your console into a pdb trace::
+        Pyramid
+        waitress
+        pyramid_debugtoolbar
+        pyramid_chameleon
 
-      def set_trace():
-          import pdb, sys
-          debugger = pdb.Pdb(stdin=sys.__stdin__, 
-              stdout=sys.__stdout__)
-          debugger.set_trace(sys._getframe().f_back)
+#. Edit ``main.py``
 
-#. `Sign up for a GAE account <http://code.google.com/appengine/>`_
-   and create an application.  You'll need a mobile phone to accept an
-   SMS in order to receive authorization.
+   Add the following lines::
```

Comment:
Please use `..code-block:: python` instead of the default `::`. IOW:

```rst
  Add the following lines.

  .. code-block:: python

      from pyramid.paster import get_app, setup_logging
      ini_path = 'production.ini'
      ini_path = 'production.ini'
      application = get_app(ini_path, 'main')
```